src/array.rs

```diff
@@ -1260,57 +1208,108 @@ impl<T: Element, D> PyArray<T, D> {
         }
     }
 
-    /// Construct a new array which has same values as self, same matrix order, but has different
-    /// dimensions specified by `dims`.
+    /// Construct a new array which has same values as self,
+    /// but has different dimensions specified by `dims`
+    /// and a possibly different memory order specified by `order`.
+    ///
+    /// Since the returned array can be identical to `self`,
```

Comment:
Is this"drop" statement worth keeping? With Python reference counting it's hard to know when the original still actually be gc'd.

Reply:
I think the situation significantly improve with dynamic borrow checking which safely handles multiple shared references to the same NumPy array. Also since `&'a PyArray<T,D>: Copy` the situation described is very easy to produce even without calling `reshape`. Hence, I agree, this can be removed.